File: Crio/crio_programming_interview_problems-master/CandyToKids/CandyToKids.py
def candyToKids(n,goodness):
# CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    goodness_index_pair= []
    for i in range(n):
        goodness_index_pair.append([goodness[i],i])
    goodness_index_pair = sorted(goodness_index_pair)
    candies = [1 for _ in goodness]
    for i in range(n):
        ind = goodness_index_pair[i][1]
        good = goodness_index_pair[i][0]
        if ind-1>=0:
            if goodness[ind-1]>good:
                candies[ind-1]=max(candies[ind-1],candies[ind]+1)
            
        if (ind+1)<n :
            if goodness[ind+1]>good:
                candies[ind+1]=max(candies[ind+1],candies[ind]+1)
    ans = sum(candies)
    return ans
# CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

def main():
    n = int(input())
    goodness = list(map(int,input().strip().split()))

    ans = candyToKids(n,goodness)
    print(ans)

if __name__=="__main__":
    main()

File: Crio/crio_programming_interview_problems-master/CandyToKids/CandyToKids.cpp
#include<bits/stdc++.h>
using namespace std;
int candyToKids(int n, vector<int>&goodness) {
// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
	int candy = 0;
	vector<pair<int, int> > candies(n);
	vector<int> actualCandies(n, 1);
	for (int i = 0; i < n; i++) {
		candies[i].first = goodness[i];
		candies[i].second = i;
	}
	sort(candies.begin(), candies.end());
	for ( int i = 0; i < n; i++) {
		pair<int, int> P = candies[i];
		if (P.second > 0 && goodness[P.second - 1] > goodness[P.second]) {
			actualCandies[P.second - 1]  = max( actualCandies[P.second - 1], actualCandies[P.second] + 1);
		}
		if ( P.second < n - 1 && goodness[P.second + 1] > goodness[P.second]) {
			actualCandies[P.second + 1] = max( actualCandies[P.second + 1] , actualCandies[P.second] + 1);
		}
	}
	for ( int i = 0; i < n; i++) {
		candy += actualCandies[i];
	}
	return candy;
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
}
int main() {
	int n, m;
	cin >> n;
	vector<int> goodness(n);
	for (int i = 0; i < n; i++) cin >> goodness[i];
	int ans = candyToKids(n, goodness);
	cout << ans << endl;
}
